include/highfive/bits/H5DataType_misc.hpp

```diff
@@ -270,7 +270,7 @@ inline size_t find_first_atomic_member_size(hid_t hid) {
 
 // Calculate the padding required to align an element of a struct
 #define _H5_STRUCT_PADDING(current_size, member_size) \
-    (((member_size) - (current_size)) % (member_size))
+    ((std::max(member_size, current_size) - std::min(member_size, current_size)) % (member_size))
```

Comment:
Good catch - the reason why this is correct and the original code wasn't, is that `%` is division remainder. So if  `(member_size) - (current_size)` is negative, then the result of the macro might be negative, messing up your padding.
(I'm mostly explaining that for my own benefit 😆 )

Reply:
These are all `size_t`, seems like a real danger of some wrap-around, too.

Reply:
you're right - but the wraparound will still break the original code - so this is a good fix.